src/lib/useOrientationChange.js

```diff
@@ -0,0 +1,22 @@
+import { useEffect } from 'react'
+import { useStore } from '../store'
+
+export function useOrientationChange() {
+  const set = useStore((state) => state.set)
+  const isMobilePortrait = useStore((state) => state.isMobilePortrait)
+  useEffect(() => {
+    const handler = () => {
+      if (window.orientation === 0) {
+        set({ isMobilePortrait: true })
+      } else {
+        set({ isMobilePortrait: false })
+      }
+    }
+    window.addEventListener('orientationchange', handler)
```

Comment:
eek- `orientationchange` is deprecated. We could use something like `resize` event.

Reply:
also don't forget the {passive: true} 😄